### PR TITLE
MAINT: Update CI, Update frameworks, Pin Deps

### PR DESCRIPTION
**Pull request by AnirudhDagar**
This PR helps move d2l-zh to the upgraded CI.

### CODE REVIEW

**Comment by astonzhang** [comment]
> @astonzhang do you think we can [not depend on a particular version of d2l lib](https://github.com/d2l-ai/d2l-zh/blob/c09aee4d007862b4c6cade66c82b841f3900b466/static/build.yml#L5) for d2l-zh? How about building it from the source? Like in [d2l-tr](https://github.com/d2l-ai/d2l-tr/blob/9a30006d1f7804b3390ca9b3c3d8d1c75663d80d/static/build.yml#L5) or [d2l-en](https://github.com/d2l-ai/d2l-en/blob/3811432b2d982f82d53a2266034e5f7e6c69db6e/static/build.yml#L5) or any other translation.

d2l-zh fixes libs since it is going to be published soon.

Can we ensure d2l-en/r0.17 https://github.com/d2l-ai/d2l-en/pull/2055 passes the whole book so I'll upload a 0.17.4 with the updated lib versions? Then we can change 0.17.3->0.17.4 here and in the installation.

cc @xiaotinghe

**Comment by AnirudhDagar** [comment]
I see, yeah that sounds like a plan. I've set up CI for d2l-zh anyway. Will update this PR once r0.17 is ready on d2l-en and 0.17.4 is uploaded.

**Comment by astonzhang** [comment]
TODO: @astonzhang 
Update installation (in the main text) with new d2l and framework versions

**Comment by astonzhang** [comment]
@d2l-bot rerun

**Comment by astonzhang** [comment]
@AnirudhDagar   can you evaluate all the notebooks (d2lbook clear) before merging this change? Thanks

**Comment by AnirudhDagar** [comment]
@d2l-bot please rerun

**Comment by AnirudhDagar** [comment]
@d2l-bot please rerun

Testing bot (2)

**Comment by AnirudhDagar** [comment]
@d2l-bot rerun

File: setup.py [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
-    'jupyter',
-    'numpy',
-    'matplotlib',
-    'requests',
-    'pandas'
+    'jupyter==1.0.0',
+    'numpy==1.22.2',
+    'matplotlib==3.4',
+    'requests==2.25.1',
+    'pandas==1.2.4'
```
Comment (xmuyong) [inline setup.py:9]: 你好，这次升级之后，没有办法在 colab  里面执行
!pip install git+https://github.com/d2l-ai/d2l-zh@release
会报错，详细信息如下：

Collecting git+https://github.com/d2l-ai/d2l-zh@release
  Cloning https://github.com/d2l-ai/d2l-zh (to revision release) to /tmp/pip-req-build-yqy6cfru
  Running command git clone -q https://github.com/d2l-ai/d2l-zh /tmp/pip-req-build-yqy6cfru
  Running command git checkout -b release --track origin/release
  Switched to a new branch 'release'
  Branch 'release' set up to track remote branch 'release' from 'origin'.
  Running command git submodule update --init --recursive -q
Requirement already satisfied: jupyter==1.0.0 in /usr/local/lib/python3.7/dist-packages (from d2l==2.0.0b0) (1.0.0)
ERROR: Could not find a version that satisfies the requirement numpy==1.22.2 (from d2l) (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0rc1, 1.13.0rc2, 1.13.0, 1.13.1, 1.13.3, 1.14.0rc1, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0rc1, 1.15.0rc2, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0rc1, 1.16.0rc2, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0rc1, 1.17.0rc2, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0rc1, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0rc1, 1.19.0rc2, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5, 1.20.0rc1, 1.20.0rc2, 1.20.0, 1.20.1, 1.20.2, 1.20.3, 1.21.0rc1, 1.21.0rc2, 1.21.0, 1.21.1, 1.21.2, 1.21.3, 1.21.4, 1.21.5)
ERROR: No matching distribution found for numpy==1.22.2


谢谢！

Comment (yangkangqi) [reply setup.py:9]: 我也是显示这样的，你解决了吗？如果解决了，方便告知一下吗

Comment (AnirudhDagar) [reply setup.py:9]: Yes this will be solved in our next release very soon. Here's the [fix](https://github.com/d2l-ai/d2l-en/pull/2073).

cc @astonzhang

**Pull request merged**